bin/database.py

```diff
@@ -1054,6 +1057,60 @@ def upgrade_to_29():
     users = config.db.users.find({})
     process_cursor(users, upgrade_to_29_closure)
 
+def upgrade_to_30_closure_analysis(coll_item, coll):
+    analyses = coll_item.get('analyses', None)
+
+    if analyses is not None:
+        for analysis_ in analyses:
+            files = analysis_.get('files', [])
+            for file_ in files:
+                if 'created' not in file_:
+                    file_['created'] = analysis_.get('created', datetime.datetime(1970, 1, 1))
+        result = config.db[coll].update_one({'_id': coll_item['_id']}, {'$set': {'analyses': analyses}})
+        if result.modified_count == 1:
+            return True
+        else:
+            return "File timestamp creation failed for:" + str(session) + '/analyses' + str(analysis_) + '/files' + str(file_)
+
+def upgrade_to_30_closure_coll(coll_item, coll):
+    files = coll_item.get('files', [])
+    for file_ in files:
+        if 'created' not in file_:
+            file_['created'] = coll_item.get('created', datetime.datetime(1970, 1, 1))
+    result = config.db[coll].update_one({'_id': coll_item['_id']}, {'$set': {'files': files}})
+    if result.modified_count == 1:
+        return True
+    else:
+        return "File timestamp creation failed for:" + str(session) + '/files' + str(file_)
+
+
+def upgrade_to_30():
+    """
+    scitran/core issue #759
+
+    give created timestamps that are missing are given based on the parent object's timestamp
+    """
+
+    cursor = config.db.collections.find({'analyses': {'$exists': True},
+                                                       'analyses.files.created': {'$exists': False}})
+    process_cursor(cursor, upgrade_to_30_closure_analysis, context = 'collections')
+
+    cursor = config.db.sessions.find({'analyses': {'$exists': True},
+                                                       'analyses.files.created': {'$exists': False}})
+    process_cursor(cursor, upgrade_to_30_closure_analysis, context = 'sessions')
+
+    cursor = config.db.sessions.find({'files': {'$exists': True}, 'files.created': {'$exists': False}})
```

Comment:
If you wanted to get a bit cheeky you could do something like:

```
for coll in ['projects', 'sessions', 'acquisitions', 'collections']:
    cursor = config.db[coll].find({'files': {'$exists': True}, 'files.created': {'$exists': False}})
    process_cursor(cursor, upgrade_to_30_closure_coll, context = coll)
    if coll in ['sessions', 'collections']:
        cursor = config.db[coll].find({'analyses': {'$exists': True}, 'analyses.files.created': {'$exists': False}})
        process_cursor(cursor, upgrade_to_30_closure_analysis, context = coll)
```
But don't feel like you need to change it. Just wanted to show a way you can take advantage of  db[<collection_name>].